### PR TITLE
[Preview] Query: Refactors EnableOptimisticDirectExecution to true by default in Preview mode

### PR DESCRIPTION
**Pull request by akotalwar**
# Pull Request Template

## Description
Enabled OptimisticDirectExecution by default in Preview mode to test its performance. Based on the results, we will decide if OptimisticDirectExecution should be opt-in or default for GA.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 1]
```diff
@@ -49,7 +49,11 @@ public class QueryRequestOptions : RequestOptions
         /// <value>
         /// Direct (optimistic) execution offers improved performance for several kinds of queries such as a single partition streaming query.
         /// </value>
-        public bool EnableOptimisticDirectExecution { get; set; }
+    #if PREVIEW
+        public bool EnableOptimisticDirectExecution { get; set; } = true;
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:53]: >bool EnableOptimisticDirectExecution { get; set; } [](http://example.com/codeflow?start=15&length=50)

For consistency, I would put this part outside #if...#endif

Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:53]: >= true; [](http://example.com/codeflow?start=66&length=7)

This needs to be in a separate if...endif

Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:53]: Please refer to ComputedProperties property on ContainerProperties class as an example.

**Comment by ealsur** [comment]
@msftbot wait for @kirankumarkolli for public API approval

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml [hunk 2]
```diff
@@ -341,10 +341,10 @@
             "RetryAfterInMs": "9000",
             "BELatencyInMs": "4.2",
             "ReplicaHealthStatuses": [
-                "http://storephysicaladdress-1p.com:Connected",
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml:344]: Why are JSON serializer setting very sensitive? 

/cc: @elasur

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/OptimisticDirectExecutionQueryBaselineTests.cs [hunk 3]
```diff
@@ -147,7 +147,7 @@ public async Task TestDefaultQueryRequestOptionsSettings()
 
             QueryRequestOptions requestOptions = new QueryRequestOptions();
 
-            Assert.AreEqual(requestOptions.EnableOptimisticDirectExecution, false);
+            Assert.AreEqual(false, requestOptions.EnableOptimisticDirectExecution);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/OptimisticDirectExecutionQueryBaselineTests.cs:150]: Same test runs for preview package as well. Isn't it supposed to be conditional?

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 4]
```diff
@@ -49,8 +49,18 @@ public class QueryRequestOptions : RequestOptions
         /// <value>
         /// Direct (optimistic) execution offers improved performance for several kinds of queries such as a single partition streaming query.
         /// </value>
-        public bool EnableOptimisticDirectExecution { get; set; }
-
+    #if PREVIEW
+        public
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:53]: By convention, # directives are not indented.

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 5]
```diff
@@ -49,8 +49,18 @@ public class QueryRequestOptions : RequestOptions
...
+    #else
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:54]: Kindly remove indentation of this line and other # directives below as well.

File: Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs [hunk 6]
```diff
@@ -49,8 +49,18 @@ public class QueryRequestOptions : RequestOptions
...
+        internal
+    #endif
+        bool EnableOptimisticDirectExecution { get; set; } = 
+            #if PREVIEW 
+                true
+            #else
+                false
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/RequestOptions/QueryRequestOptions.cs:61]: >false [](http://example.com/codeflow?start=16&length=5)

By convention, we don't assign the default (false) values.
So following should suffice:
#if PREVIEW
  public
#else
  internal
#endif
  bool ... {get;set;}
#if PREVIEW
  = true;
#endif

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml [hunk 7]
```diff
@@ -341,10 +341,10 @@
...
-                "http://storephysicaladdress-2s.com:Unknown",
-                "http://storephysicaladdress-3s.com:Unhealthy",
-                "http://storephysicaladdress-4s.com:Unknown"
+              "http://storephysicaladdress-1p.com:Connected",
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml:344]: How is this diff related to your chage?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/OptimisticDirectExecutionQueryBaselineTests.cs [hunk 8]
```diff
@@ -144,10 +144,15 @@ public void NegativeOptimisticDirectExecutionOutput()
         [TestMethod]
         public async Task TestDefaultQueryRequestOptionsSettings()
         {
-
             QueryRequestOptions requestOptions = new QueryRequestOptions();
-
-            Assert.AreEqual(requestOptions.EnableOptimisticDirectExecution, false);
+            bool odeExpectedValue =
+                #if PREVIEW
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/OptimisticDirectExecutionQueryBaselineTests.cs:149]: Please remove the indentation. For remaining directives as well.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml (hunk 7 above)
Comment (akotalwar) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml:344]: Due to a change in the public contract, I had to run a script called UpdateContracts.ps1. It was this script that made this change.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/TraceWriterBaselineTests.TraceData.xml:344]: Sounds like space changes? Could it be due to machine configuration?

**Review by kirankumarkolli — APPROVED** [review]
